### ISSUE TITLE
Custom Renderer for Vertical Lines

### ISSUE THREAD

**Issue by GuiPretto** [namespace-ee/react-calendar-timeline#793]
**Is your feature request related to a problem? Please describe.**
I want to highlight the interval where the TodayMarker is on every n time. 
I know that right now we can add classes at creation time, but I would like to change classes (re-render) when, for example, the TodayMarker leaves 10h interval and enters 11h interval.

**Describe the solution you'd like**
Something just like itemRenderer, groupRenderer and intervalRenderer, but for verticalLines.

**Comment by rosanzheng** [comment]
Hey,
There is a prop called verticalLineClassNamesForTime - Check the documentation for usage.
But I must say, I tried using it and I am having quite some struggles with it as an error occurs, preventing the code from compiling.
I already submitted an issue #794 - You can take a look at the code here: https://codesandbox.io/s/timeline-demo-error-verticallineclassnamesfortime-of8u9

If you detect any errors from my side, I'd appreciate it if you can help me out with that!

**Comment by GuiPretto** [comment]
Hi!! Thanks for replying here :).
First of all, I solved my problem, and I'm gonna tell later how I did it, but let me explain better what my problem was first.

I knew about verticalLineClassNamesForTime, and I used it, but the problem is that it only worked on the build of the timeline, and not after every update. So the classes I wanted were added, but not changed after specific time. So, I needed to refresh the page to see the correct highlight.

So, to solve it, I just used useCallback in the function, using as dependency what I needed to be the re-render factor. In the example below I used a state called 'actualTime', which updates every 1s. This way, the function was updated and the timeline re-rendered every 1s.

```javascript
  const verticalLine = useCallback((start, end) => {
    const day = moment().format('D')
    const intervalDay = moment(start).format('D')
    if (intervalDay == day) {
      if (actualTime >= start && actualTime < end) {
        return ['actual-interval']
      } else {
        return []
      }
    } else {
      return ['blank-interval']
    }
  }, [actualTime])
```
